colour/colorimetry/spectrum.py

```diff
@@ -1939,8 +2050,49 @@ def align(self,
                 77.18     ...,  77.18     ...,  77.18     ...,  77.18     ...])
         """
 
-        self.interpolate(shape)
-        self.extrapolate(shape, method, left, right)
+        self.interpolate(shape, interpolation_method)
+        self.extrapolate(shape,
+                         extrapolation_method,
+                         extrapolation_left,
+                         extrapolation_right)
+
+        return self
+
+    def trim(self, shape):
```

Comment:
Oh ok! You were thinking about something like that: http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.trim_zeros.html

Any alternative names for that? `cull`? http://www.thesaurus.com/browse/cull?s=t